### ISSUE TITLE
Show docstrings on keys of TypedDicts

### ISSUE THREAD

**Issue by rattrayalex** [microsoft/pylance-release#2508]
## Current behavior:

Docstring is not visible for the key:

<img width="709" alt="Screen Shot 2022-03-24 at 4 27 39 PM" src="https://user-images.githubusercontent.com/704302/160004661-07a1fec0-5fbf-4b03-b4f6-731783869e07.png">

<details>
<summary>Python Code</summary>

```py
class Circle(TypedDict):
  radius: int
  """The radius of the circle in pixels."""

  x: int
  """The horizontal position on the screen in pixels."""

  y: int
  """The vertical position on the screen in pixels."""

def show_circle(circle: Circle) -> None:
  print(circle)

show_circle({"radius": 4, "x": 3, "y": 2})
```

</details>

## Desired behavior:

Show the docstring for the dict key, as it does in TypeScript:

<img width="623" alt="Screen Shot 2022-03-24 at 4 29 59 PM" src="https://user-images.githubusercontent.com/704302/160005043-58a7d5c7-13fa-4ff1-b486-b1b79d9ad5a1.png">

<details>
<summary>TS Code</summary>

```ts
type Circle = {
  /** The radius of the circle in pixels. */
  radius: number;

  /** The horizontal position on the screen in pixels. */
  x: number;

  /** The vertical position on the screen in pixels. */
  y: number;
}

function showCircle(circle: Circle): void {
  console.log(circle)
}

showCircle({"radius": 4, "x": 3, "y": 2})
```
</details>

## Motivating use-case

I am designing an API client library, and want to be able to provide a good experience for code like this:

```py
my_api.customers.update(customer.id, {
  "status": "active" # Hovering over "status" should describe what the various options mean.
})
```

**Comment by erictraut** [comment]
Have you considered using a dataclass rather than a TypedDict within your API? Typing and language server support is generally quite a bit better for dataclass.

**Comment by rattrayalex** [comment]
Hi Eric,

Thanks, I'm certainly open to considering it. As a matter of fact, we started out with pydantic models. My concern is that it would be a clunky/unpythonic API for users to have to do things like this:

```py
from myapi import Client, CustomerCreateParams

client = myapi.Client()

params = CustomerCreateParams(
  name="Eric Traut",
  employer=CustomerCreateParams.Employer(
    name="Microsoft"
    address=CustomerCreateParams.Employer.Address(
      line1="123 Main St",
      # …
    ),
  )
  
)
client.customers.create(params)
```

It is my impression that most pythonistas would much prefer this:

```py
from myapi import Client

client = Client()

client.customers.create({
  "name": "Eric Traut",
  "employer": {
    "name": "Microsoft",
    "address": {
      "line1": "123 Main St",
    }
  }
})
```

However, I have not actively developed in Python for many years, so I may be off-base – your feedback would be much appreciated. (For example, there may be a cleaner way of expressing concepts like `myapi.CustomerCreateParams.Employer.Address`).

**Comment by debonte** [comment]
Reopening so we remember to include this in Pylance's release notes next week. We'll close this issue when the fix ships in Pylance.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.4.0, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202240-6-April-2022)
